cmake/IgnUtils.cmake

```diff
@@ -1804,3 +1804,366 @@ macro(_ign_cmake_parse_arguments prefix options oneValueArgs multiValueArgs)
+
+macro(ign_export_plugin path_to_install_destination)
+
+  list(APPEND plugins_path ${ign_export_plugin_INSTALL_DESTINATION})
+
+endmacro()  
+
+#####################################################
+#ign_add_executables(path_to_executables)
+#Installs all the executables inside the folder using various arguments from the user
+#Other than common link_libraries and directories,you can simply add specific executable dependencies one by one after this macro
+#The deafult file type is .cc for executables,you can add other file types using EXECUTABLES_EXTENSION argument
+
+macro(ign_add_executables path_to_executables)
+
+  set(oneValueArgs INSTALL_DESTINATION)
+  set(multiValueArgs COMMON_PUBLIC_LIBRARIES COMMON_PRIVATE_LINK_LIBRARIES COMMON_PUBLIC_DIRECTORIES COMMON_PRIVATE_LINK_DIRECTORIES EXECUTABLE_EXTENSION)
+
+  _ign_cmake_parse_arguments(ign_add_executable "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+  if(NOT ign_add_executable_EXECUTABLE_EXTENSION)
+    list(APPEND ign_add_executable_EXECUTABLE_EXTENSION ".cc")
+  endif()
+
+  foreach(EXTENSION ${ign_add_executable_EXECUTABLE_EXTENSION})
+
+    file(GLOB source_list_executable_${EXTENSION} CONFIGURE_DEPENDS
+    "${path_to_executables}/*${EXTENSION}"
+    )
+
+    list(APPEND source_list_executables ${source_list_executable_${EXTENSION}})
+
+  endforeach() 
+
+  foreach(executable_PATH ${source_list_executables})
+    get_filename_component(executable_NAME "${executable_PATH}" NAME_WLE)
+    add_executable(${executable_NAME} ${executable_PATH})
+
+    set_property(TARGET ${executable_NAME} PROPERTY CXX_STANDARD 17)
+
+    target_link_libraries(${executable_NAME}
+      PUBLIC
+        ${ign_add_executable_COMMON_PUBLIC_LIBRARIES}
+      PRIVATE
+        ${ign_add_executable_COMMON_PRIVATE_LINK_LIBRARIES}
+    )
+    target_include_directories(${executable_NAME}
+      PUBLIC
+        ${ign_add_executable_COMMON_PUBLIC_DIRECTORIES}
+      PRIVATE
+        ${ign_add_executable_COMMON_PRIVATE_LINK_DIRECTORIES}
+    ) 
+
+    if(ign_add_executable_INSTALL_DESTINATION)
+      install(
+      TARGETS ${executable_NAME}
+      DESTINATION ${ign_add_executable_INSTALL_DESTINATION})
+
+    else()
+      install(
+      TARGETS ${executable_NAME}
+      DESTINATION bin)
+
+    endif()
+
+  endforeach()
+  
+endmacro()
+#################################################################################
+#ign_add_msgs(path_to_msgs)
+#Installs all the msgs inside the folder using various arguments from the user
+#Also provides the path to ign_environment_hook()
+#Other than pre coded dependencies,you can simply add specific message dependencies one by one after this macro
+#The deafult file type is .proto for msgs,you can add other file types using MSG_EXTENSION argument
+
+macro(ign_add_msgs path_to_msgs )
+
+  set(oneValueArgs INSTALL_DESTINATION SCOPE)
+  set(multiValueArgs MSG_EXTENSION COMMON_LANGUAGE COMMON_IMPORT_DIRS COMMON_PROTOC_OUT_DIR)
```

Comment:
Whats with the "common" being used here. What are "common" dependencies? How is the user to know what "common" dependencies mean?